stdlib/lang-parser/parserModule/parser.bal

```diff
@@ -0,0 +1,1260 @@
+            if (self.LAToken(1) == LEXER_ERROR_TOKEN) {
+                log:printError(self.LookaheadToken(1).lineNumber + ":" + self.LookaheadToken(1).startPos + ": expected " +
+                tokenNames[mToken] + ";found an invalid token '" + self.LookaheadToken(1).text + "'");
+            } else {
+                log:printError(self.LookaheadToken(1).lineNumber + ":" + self.LookaheadToken(1).startPos + ": expected " +
+                tokenNames[mToken] + "; found '" + self.LookaheadToken(1).text + "'");
+            }
+
+            Token panicToken = self.panicRecovery(mToken, rule);
+            return panicToken;
+        }
+    }
+
+    #error recovery : token  insertion
+    # the expected token will be inserted inplace of the mismatched token
+    # +return - Token
+    function insertToken(int mToken) returns Token {
+        log:printError(tokenNames[mToken] + " inserted");
+        return {
+            tokenType: mToken,
+            text: tokenNames[mToken],
+            startPos: -1,
+            endPos: -1,
+            lineNumber: 0,
+            index: -1,
+            whiteSpace: ""
+        };
+    }
+
+    #error recovery : delete token
+    # mismatched token is consumed and added to the error token list
+    # +return - Token
+    function deleteToken() returns Token {
+        Token invalidToken = self.parserBuffer.consumeToken();
+        log:printError(invalidToken.lineNumber + ":" + invalidToken.startPos + ": invalid token '" +
+        invalidToken.text + "'");
+        self.errTokens[self.errCount] = invalidToken;
+        self.errCount += 1;
+        self.invalidOccurence = true;
+        return invalidToken;
+    }
+
+    #panic recovery
+    #each grammar rule will have a specific error recovery.
+    #here all the token will be removed and added to the error token list until we meet a terminal token
+    # +return - Token
+    function panicRecovery(int mToken, NodeKind rule) returns Token {
+        boolean panicMode = true;
+        self.errorRecovered = false;
+
+        if (rule == "variableDefinitionStatement") {
+            if (mToken == SEMICOLON) {
+                Token insertSemi = self.insertToken(mToken);
+                insertSemi.text = ";";
+
+                return insertSemi;
+            } else if (mToken == ASSIGN) {
+                Token insertAssign = self.insertToken(mToken);
+                insertAssign.text = "=";
+
+                return insertAssign;
+            } else {
+                int[] exprPanic = [ASSIGN, SEMICOLON, RBRACE, EOF];
+                while (panicMode) {
+                    if (self.LAToken(1) == exprPanic[0]) {
+                        break;
+                    } else if (self.LAToken(1) == exprPanic[1]) {
+                        break;
+                    } else if (self.LAToken(1) == exprPanic[2] && self.LookaheadToken(1).whiteSpace == "\n") {
+                        break;
+                    } else if (self.LAToken(1) == exprPanic[3]) {
```

Comment:
Shall we combine these if/else if blocks to a single if block, since we are doing the same thing on any one of them being true?